### PR TITLE
Do more linting and testing on TravisCI

### PR DESCRIPTION
**Pull request by islemaster**
Create a `rake travis` task that runs a more comprehensive set of tests on our code.  Travis should now do the following (in this order).
- Lint
  - Ruby
  - HAML
  - apps (runs `grunt jshint`)
- Test
  - apps (runs `grunt mochaTest`)
  - blockly-core
  - dashboard
  - pegasus

Before it was only running the "test dashboard" step.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -19,16 +19,33 @@ def create_database(uri)
...
+  
+  task :apps do
+    Dir.chdir(apps_dir) do
+      HipChat.log 'Linting <b>apps</b>...'
+      RakeUtils.system 'grunt jshint'
+    end
+  end
 
-  task all: [:ruby, :haml]
+  task all: [:ruby, :haml, :apps]
```
Comment (joshlory) [inline Rakefile]: Also `/shared`?

**Comment by islemaster** [comment]
Brent and Josh want to know if we can break out parts of our builds/tests.  We can!  I'll totally try this: http://docs.travis-ci.com/user/speeding-up-the-build/

File: Rakefile (hunk 1 above)
Comment (islemaster) [inline Rakefile]: Oh, didn't see that we have `gulp lint` set up for shared.  On it.

**Comment by islemaster** [comment]
First Travis builds failed due to syntax error in Rakefile :cry: fixing and trying again...

**Comment by islemaster** [comment]
Another big possible optimization - exclude dashboard tests from our test matrix if no files under /dashboard were touched in the changeset.

**Comment by islemaster** [comment]
I hereby declare this experiment a massive failure.  I can't get the dashboard tests to run from within the rake task, so even existing functionality doesn't work, and I can't lint or test our javascript because I can't `npm install` because the for some reason Travis' version of npm can't handle local dependencies in package.json.

(╯°□°）╯︵ ┻━┻

**Pull request closed without merging**